src/cmds/net/telnetd/telnetd.c

```diff
@@ -368,10 +393,54 @@ int main(int argc, char **argv) {
 	int telnet_connections_count;
 
 	if (argc > 1) {
-		client_descr = atoi(argv[1]);
-		MD(printf("telnetd: %d\n", client_descr));
-		telnetd_client_handler((void *)client_descr);
-		_exit(0);
+		struct client_args client_args;
+
+		if (0 == strcmp("connection", argv[1])) {
+#if !PTHEAD_USE
```

Comment:
It's probably better to have two functions here - for pthreads and without pthreads, since wth these ifdef's it's hard to understand what happens.